cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManagerImpl.java

```diff
@@ -205,9 +207,19 @@ public void close() {
      */
     @Override
     public String ensureIndexed(List<FieldSort> fieldNames) throws QueryException {
+        List<Index> indexes = this.listIndexes();
+        Collections.sort(fieldNames);
+        for(Index index: indexes){
```

Comment:
Doesn't account for the `indexType` or `tokenize` options on the Index. I wonder if it's easier to select the already existed index(es) directly with a SQL query.